src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,278 @@
+        const positiveSymbol = this.positiveSign ? '+' : ''
+
+        this.signSymbol = this.options.positiveOnly ? positiveSymbol
+          : negative ? '-' : positiveSymbol
+      }
+    },
+    numeralPrefix () {
+      return this.options.prefix ? this.options.prefix : ''
+    },
+    numeralSuffix () {
+      return this.options.suffix ? this.options.suffix : ''
+    }
+  },
+
+  methods: {
+    unmaskNumeralText (text) {
+      if (!text) return text
+
+      this.sign = text = String(text)
+      text = this.internalChange ? this.removeNonNumeral(text) : this.attemptNumeralCorrection(text)
+      if (!text) return text // attemptNumeralCorrection() may return null or falsy
+
+      let number
+      if (this.precision) {
+        number = text.split('')
+        number.splice(-this.precision, 0, '.')
+        number = this.parseFloat(number)
+      } else {
+        number = this.parseInt(text)
+      }
+
+      return this.sign + number.toString()
+    },
+    maskNumeralText (text) {
+      if (!text) return this.numeralZero()
+
+      const blockSize = Array.isArray(this.blockSize) ? this.blockSize : [this.blockSize]
+      const masked = []
+
+      let i = this.blockSeparator.length - 1
+      let j = blockSize.length - 1
+      let length = 0
+      let integer = !this.precision
+
+      text = this.sign ? String(text).substr(1) : String(text)
+
+      // Group the digits with specified separators.
+      // The first size and separator element
+      // of the arrays becomes the default value.
+      for (const digit of text.split('').reverse()) {
+        digit !== '.' && masked.push(digit)
+        if (integer && !(++length % blockSize[j])) {
+          masked.push(this.oneChar(this.blockSeparator[i], ' '))
+          i > 0 && i--
+          j > 0 && j--
+          length = 0
+        } else if (digit === '.') {
+          masked.push(this.decimal)
+          integer = true
+        }
+      }
+
+      masked.reverse()
+      // Remove leading block separator
+      this.blockSeparator.includes(masked[0]) && masked.shift()
+
+      return this.addPrefixSuffix(masked.join(''))
+    },
+    numeralZero () {
+      const zero = 0
+      this.sign = '+'
+      return this.addPrefixSuffix(zero.toFixed(this.precision).toString().replace('.', this.decimal))
+    },
+    oneChar (char, singleChar = '.') {
+      return char ? char[0] : singleChar
+    },
+    absVal (value) {
+      let parsed = parseInt(value, 10)
+
+      if (isNaN(parsed)) return 0
+      parsed = Math.abs(parsed)
+
+      // Set the general limit to that of Number.prototype.toFixed()
+      return parsed > 20 ? 20 : parsed
+    },
+    adjustNumeralCaret (selection, text) {
+      // Caret is at prefix
+      if (selection === 0) return this.numeralPrefix.length + this.sign.length
+
+      // Whether it's an integer or a fractional where abs(value) >= 10
+      if (!this.precision ||
+        (String(this.lazyValue).length > this.precision + (this.sign ? 3 : 2))) {
```

Comment:
What is the meaning of this 2/3 constants?

Reply:
The right side of the equality uses `this.precision`.
When there's a sign symbol, we need to add 1 on the right side to match the length in `this.lazyValue`.
When the value equals or is more than 10.0, we need to add 2 on the right side to include 1.0 or 2.0 etc. to match the length of `this.lazyValue`.

All this is because the fall through codes only adjust the caret in situations when we need to maintain a minimum digit count. That is, it could be between 0.0000 and 9.9999. Otherwise, 0.0000 would become 0.